backends/bmv2/common/sharedActionSelectorCheck.h

```diff
@@ -33,10 +35,11 @@ using SelectorInput = std::vector<const IR::Expression *>;
 // This pass makes sure that when several match tables share a selector, they use the same input for
 // the selection algorithm. This is because bmv2 considers that the selection key is part of the
 // action_selector while v1model.p4 considers that it belongs to the table match key definition.
+template <Standard::Arch arch>
 class SharedActionSelectorCheck : public Inspector {
+    BMV2::ConversionContext* ctxt;
```

Comment:
Mostly moved from cpp, changed constructor to take ConversionContext, instead of refmap and typemap. Refmap and typemap are taken from the pointer inside ConversionContext. Same implementation as before.